src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs

```diff
@@ -4395,6 +4395,126 @@ public static void DerivedTypeWithDifferentOverrides()
     }
 #endif
 
+    //This case currently will fail in ReflectionOnly implementation when targeting uapaot
+#if !uapaot || !ReflectionOnly 
+    [Fact]
+    public static void Xml_DefaultNamespaceChange_SimpleArray_ObjectAsRoot()
+    {
+        SimpleType[] x = new SimpleType[] { new SimpleType { P1 = "abc", P2 = 11 }, new SimpleType { P1 = "def", P2 = 12 } };
+#if uapaot && !ReflectionOnly
+        Func<XmlSerializer> serializerFactory = () => { return new XmlSerializer(typeof(object)); };
```

Comment:
@JimboWei I thought about this. I think we should actually remove the above usage, and just keep the usage in `else`. It should work across all configurations. Can you please remove the above two `#if`? Thanks.